### PR TITLE
SYS-164 added workflow to run linter and trigger smoke tests

### PR DESCRIPTION
**Pull request by TieDyedSheep**
Added a github action yaml to to basic builds, linting, test the patches can be applied and finally trigger the Jenkins smoke test job. If no patches exist, the job will skip that step. if the build, linting, formatter or patches fail, the workflow will fail.

### CODE REVIEW

**Review by muni-corn — DISMISSED** [review]

**Comment by muni-corn** [comment]
Looks good to me!

**Comment by TieDyedSheep** [comment]
This is the Jenkins job I trigger, it was provided parameters to override the shardus_types parameters with the branch from the PR and this triggered it's own action and ran successfully
https://devs-work.shardeum.org/job/Testing/job/Smoke/job/smoke_10_plus_10/113/
https://github.com/shardeum/lib-types/actions/runs/9103671559

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
```
Comment (chrypnotoad) [inline .github/workflows/ci.yml]: our repos use `18.16.1`

**Review by chrypnotoad — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (TieDyedSheep) [reply .github/workflows/ci.yml]: switched this to 18 and retested

**Comment by TieDyedSheep** [comment]
This was approved and the approval was removed when I resolved a conversation so I'm just bypassing and merging this

**Pull request merged**